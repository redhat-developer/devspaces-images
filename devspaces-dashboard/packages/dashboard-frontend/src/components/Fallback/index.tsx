/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import { Spinner, spinnerSize } from '@patternfly/react-core';
import React from 'react';

const Fallback = (
  <div style={{ height: '100%', textAlign: 'center', opacity: '0.5' }}>
    <Spinner
      data-testid="fallback-spinner"
      size={spinnerSize.xl}
      style={{ top: 'calc(50% - 18px)' }}
    />
  </div>
);

export default Fallback;
