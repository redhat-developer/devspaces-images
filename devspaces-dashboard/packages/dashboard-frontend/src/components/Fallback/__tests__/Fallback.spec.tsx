/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';
import renderer, { ReactTestRendererJSON } from 'react-test-renderer';
import Fallback from '..';

describe('React element that is rendered while waiting for the component to load', () => {
  it('should render fallback correctly', () => {
    expect(getComponentSnapshot(Fallback)).toMatchSnapshot();
  });
});

function getComponentSnapshot(
  component: React.ReactElement,
): null | ReactTestRendererJSON | ReactTestRendererJSON[] {
  return renderer.create(component).toJSON();
}
