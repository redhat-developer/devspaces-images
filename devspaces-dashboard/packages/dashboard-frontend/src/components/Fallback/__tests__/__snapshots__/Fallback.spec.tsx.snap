// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React element that is rendered while waiting for the component to load should render fallback correctly 1`] = `
<div
  style={
    Object {
      "height": "100%",
      "opacity": "0.5",
      "textAlign": "center",
    }
  }
>
  <span
    aria-valuetext="Loading..."
    className="pf-c-spinner pf-m-xl"
    data-testid="fallback-spinner"
    role="progressbar"
    style={
      Object {
        "top": "calc(50% - 18px)",
      }
    }
  >
    <span
      className="pf-c-spinner__clipper"
    />
    <span
      className="pf-c-spinner__lead-ball"
    />
    <span
      className="pf-c-spinner__tail-ball"
    />
  </span>
</div>
`;
